### PR TITLE
Only Include Required Modules in the Generated From Source Aggregate POM 

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR introduces a new script which generates an aggregate POM for the "From Source" test runs. The new script only includes modules which are required during the "From Source" run instead of building all Track 2 libraries as the previous generated POM did. With this change the "From Source" runs should reduce their run times by 2-5 minutes (depending on how many projects depend on the libraries in the project list).

### CODE REVIEW

File: eng/scripts/generate_from_source_pom.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+# Beginning XML for the aggregate POM.
+pom_file_start = '''<!-- Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT License. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-sdk-from-source</artifactId>
+  <packaging>pom</packaging>
+  <version>1.0.0</version> <!-- Need not change for every release-->
```
Comment (srnagar) [inline eng/scripts/generate_from_source_pom.py:46]:
```suggestion
  <version>1.0.0</version>
```

File: eng/scripts/generate_from_source_pom.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    start_time = time.time()
+    create_from_source_pom(args.project_list)
+    elapsed_time = time.time() - start_time
+
+    print('Effective From Source POM File')
+    with open(file=client_from_source_pom_path, mode='r') as fromSourcePom:
+        print(fromSourcePom.read())
+
+    print('elapsed_time={}'.format(elapsed_time))
+    print('Total time for replacement: {}'.format(str(timedelta(seconds=elapsed_time))))
```
Comment (srnagar) [inline eng/scripts/generate_from_source_pom.py:236]:
```suggestion
    print('Total time for replacement: {} seconds'.format(str(timedelta(seconds=elapsed_time))))
```

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 3]
```diff
@@ -311,7 +311,12 @@ jobs:
...
 
-      - task: PowerShell@2
+      - template: ../steps/generate-project-list.yml
+        parameters:
+          Artifacts: ${{parameters.Artifacts}}
+          AdditionalModules: ${{parameters.AdditionalModules}}
+
+      - task: PythonScript@0
         # The root pom.xml contains both track 1, data, and track 2, client libraries.
         # This task calls a powershell script that will go through all of the pom files
```
Comment (srnagar) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:321]:
```suggestion
        # This task calls a Python script that will go through all of the pom files
```

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 4]
```diff
@@ -320,16 +325,10 @@ jobs:
         # jacoco and spotbugs which are unnecessary for the From Source run.
         displayName: 'Generate Aggregate Pom for From Source Build'
         inputs:
-          pwsh: true
-          workingDirectory: $(Agent.BuildDirectory)
-          filePath: eng/scripts/Generate-Client-Aggregate-Pom.ps1
```
Comment (srnagar) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:325]: Can this script be deleted now? - [Generate-Client-Aggregate-Pom](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/scripts/Generate-Client-Aggregate-Pom.ps1)

**Review by srnagar — APPROVED** [review]

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 4 above)
Comment (alzimmermsft) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:325]: Most likely, I'd leave it around for a while (month or so) while this change settles and is validated completely

**Pull request merged**